packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3782,55 +3785,91 @@ class _Editable extends MultiChildRenderObjectWidget {
+
+/// A function that takes the Offset at which focus is requested.
+typedef _ScribbleFocusCallback = void Function(Offset offset);
+
+/// A widget that provides the ability to receive handwriting input from
+/// [Scribble].
 class _ScribbleFocusable extends StatefulWidget {
   const _ScribbleFocusable({
     required this.child,
-    required this.focusNode,
-    required this.editableKey,
-    required this.updateSelectionRects,
     required this.enabled,
+    required this.focusNode,
+    required this.onPlaceholderLocationChanged,
+    required this.onScribbleFocus,
+    required this.onShowToolbar,
+    required this.readOnly,
+    required this.value,
   });
 
   final Widget child;
-  final FocusNode focusNode;
-  final GlobalKey editableKey;
-  final VoidCallback updateSelectionRects;
   final bool enabled;
+  final FocusNode focusNode;
+  final _PlaceholderLocationCallback onPlaceholderLocationChanged;
+  final _ScribbleFocusCallback onScribbleFocus;
+  final VoidCallback onShowToolbar;
+  final bool readOnly;
+  final TextEditingValue value;
 
   @override
   _ScribbleFocusableState createState() => _ScribbleFocusableState();
 }
 
-class _ScribbleFocusableState extends State<_ScribbleFocusable> implements ScribbleClient {
+class _ScribbleFocusableState extends State<_ScribbleFocusable> with ScribbleClient {
   _ScribbleFocusableState(): _elementIdentifier = (_nextElementIdentifier++).toString();
 
+  void _onFocusChange() {
+    _updateClient(widget.focusNode.hasFocus);
+  }
+
+  void _updateClient(bool hasFocus) {
+    if (hasFocus && Scribble.client != this) {
+      Scribble.client = this;
+    } else {
+      Scribble.client = null;
+    }
+  }
+
   @override
   void initState() {
     super.initState();
+    _updateClient(widget.focusNode.hasFocus);
+    widget.focusNode.addListener(_onFocusChange);
     if (widget.enabled) {
-      TextInput.registerScribbleElement(elementIdentifier, this);
+      Scribble.registerScribbleElement(elementIdentifier, this);
     }
   }
 
   @override
   void didUpdateWidget(_ScribbleFocusable oldWidget) {
     super.didUpdateWidget(oldWidget);
+    if (oldWidget.focusNode != widget.focusNode) {
+      oldWidget.focusNode.removeListener(_onFocusChange);
+      widget.focusNode.addListener(_onFocusChange);
+      _updateClient(widget.focusNode.hasFocus);
+    }
     if (!oldWidget.enabled && widget.enabled) {
-      TextInput.registerScribbleElement(elementIdentifier, this);
+      Scribble.registerScribbleElement(elementIdentifier, this);
     }
 
     if (oldWidget.enabled && !widget.enabled) {
-      TextInput.unregisterScribbleElement(elementIdentifier);
+      Scribble.unregisterScribbleElement(elementIdentifier);
     }
   }
 
   @override
   void dispose() {
-    TextInput.unregisterScribbleElement(elementIdentifier);
+    Scribble.unregisterScribbleElement(elementIdentifier);
+    widget.focusNode.removeListener(_onFocusChange);
+    Scribble.client = null;
```

Comment:
what if this client didn't have the focus?

Reply:
Great catch here and your next comment.  I've updated it here and there to only set the client to null if this instance is currently the client.